docs/_pages/typecheck.md

```diff
@@ -351,6 +351,30 @@ local onlyString: string = stringOrNumber -- ok
 local onlyNumber: number = stringOrNumber -- not ok
 ```
 
+## Typecasts
+
+Expressions may be typecast using `::`.  Typecasting is useful for specifying the type of an expression when the automatically inferred type is too generic.
+
+For example, consider the following table constructor where the intent is to store a table of names:
+```lua
+local myTable = {names = {}}
+table.insert(myTable.names, 42)         -- Inserting a number ought to cause a type error, but doesn't
+```
+
+In order to specify the type of the `names` table a typecast may be used: 
+
+```lua
+local myTable = {names = {} :: {string}}
+table.insert(myTable.names, 42)         -- not ok, invalid 'number' to 'string' conversion
+```
+
+A typecast itself is also type checked to ensure the conversion specified is a valid one:
+```lua
+local number = 1
+local value = number :: any             -- ok, 'number' is a subtype of 'any'
+local flag = number :: boolean          -- not ok, invalid 'number' to 'boolean' conversion
```

Comment:
Using number that is also a type name as a local name here was a bit confusing too me :)

Reply:
I merged before your comments, but I'll create a PR to fix this.  I think renaming the variable to `numericValue` should avoid the confusion.

Reply:
Well, this is a minor note, so might not warrant a separate PR by itself.